state-sync/storage-service/server/src/tests/mock.rs

```diff
@@ -378,6 +379,22 @@ pub fn create_mock_db_for_optimistic_fetch(
     db_reader
 }
 
+/// Creates a mock db with the basic expectations required to handle optimistic fetch requests
+pub fn create_mock_db_for_optimistic_fetch(
+    highest_ledger_info_clone: LedgerInfoWithSignatures,
+    lowest_version: Version,
+) -> MockDatabaseReader {
+    create_mock_db_for_storage_summary_updates(highest_ledger_info_clone, lowest_version)
+}
+
+/// Creates a mock db with the basic expectations required to handle subscription requests
+pub fn create_mock_db_for_subscription(
+    highest_ledger_info_clone: LedgerInfoWithSignatures,
+    lowest_version: Version,
+) -> MockDatabaseReader {
+    create_mock_db_for_storage_summary_updates(highest_ledger_info_clone, lowest_version)
```

Comment:
copy-paste-o? this looks the same as the other one?

Reply:
Aah, this was by design, to make the tests more readable (i.e., when comparing the optimistic fetch and subscription tests and mocks). But, I've now merged the two and exposed them as `create_mock_db_with_summary_updates`.